readthedocs/core/views/hooks.py

```diff
@@ -136,91 +137,135 @@ def _build_url(url, projects, branches):
 
 @csrf_exempt
 def github_build(request):
-    """A post-commit hook for github."""
+    """GitHub webhook consumer
+
+    This will search for projects matching either a stripped down HTTP or SSH
+    URL. The search is error prone, use the API v2 webhook for new webhooks.
+    """
     if request.method == 'POST':
         try:
-            # GitHub RTD integration
-            obj = json.loads(request.POST['payload'])
```

Comment:
Actually, that would mean GitHub was sending the payload as JSON encoded in the `payload` field?  I don't see any signs that this was the case.